src/ast-utils.ts

```diff
@@ -114,6 +114,28 @@ export function parse(html: string, options: ParserOptions): ASTNode {
 }
 
 /**
+ * Returns true if the nodes are given in order as they appear in the source
+ * code.
+ */
+export function inOrder(left: ASTNode, right: ASTNode): boolean {
+  const l = left.__location, r = right.__location;
+  return l && r && l['line'] && r['line'] &&
```

Comment:
Apparently not.  I could add typings for `LocationInfo` and `ElementLocationInfo` but I guess I'd rather do that after porting this to `dom5`.  I'll add a TODO.